### PR TITLE
Warning about too long input for fast tokenizers too

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

If truncation is not set in tokenizers, but the tokenization is too long 
for the model (`model_max_length`), we used to trigger a warning that

The input would probably fail (which it most likely will).

This PR re-enables the warning for fast tokenizers too and uses common code
for the trigger to make sure it's consistent across.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors which may be interested in your PR.

@LysandreJik
@thomwolf

### CODE REVIEW

**Comment by Narsil** [comment]
Failing tests seem to come from some other code (seq2seq)

File: src/transformers/tokenization_utils_base.py [hunk 1]
```diff
@@ -3170,13 +3163,11 @@ def get_special_tokens_mask(
...
-        Clean up a list of simple English tokenization artifacts like spaces before punctuations and abbreviated forms.
+        clean up a list of simple english tokenization artifacts like spaces before punctuations and abbreviated forms.
 
-        Args:
-            out_string (:obj:`str`): The text to clean up.
+        args: out_string (:obj:`str`): the text to clean up.
 
-        Returns:
-            :obj:`str`: The cleaned-up string.
+        returns: :obj:`str`: the cleaned-up string.
```
Comment (LysandreJik) [inline src/transformers/tokenization_utils_base.py:3170]: The docstring was in the correct style before the changes

**Comment by LysandreJik** [comment]
@thomwolf could you review this PR as you're the mastermind behind this code?

**Review by thomwolf — APPROVED** [review]
LGTM

File: src/transformers/tokenization_utils_base.py (hunk 1 above)
Comment (Narsil) [reply src/transformers/tokenization_utils_base.py:3170]: But I simply ran the documentation fixer :(

**Comment by Narsil** [comment]
@LysandreJik May I merge (failing tests and quality is linked to unrelated `finetune.py` code, I tried to rebase but it does not seem to be enough)

**Review by LysandreJik — APPROVED** [review]

**Pull request merged**